### PR TITLE
Add other events to views

### LINKED ISSUE #49: Add otherEvents to events views
We need to add `other_events` to:

- Schedule view
- Event details view

### PR DESCRIPTION
**Pull request by fourlastor**
## Problem

Closes #49  - This PR projects the table `other_events` on the views `schedule` and `event_details`.

## Solution

It's time to remove duplication! The code was literally a copy paste and finally I got the chance to make some order, extract some common functions and clean up the two endpoints.

They now live in a common package: `events-view` and share a huge percentage of the code.

### CODE REVIEW

File: functions/src/events-view/events-view-data.ts [hunk 1]
```diff
@@ -18,11 +18,11 @@ export interface Event {
     startTime: Date
     endTime: Date
     place: Place | null
-    track: Track | null
-    speakers: Speaker[]
-    experienceLevel: string | null
     type: string
-    description: string | null
```
Comment (fourlastor) [inline functions/src/events-view/events-view-data.ts]: small detail here: the question mark indicates that the value can not be there at all, instead of being undefined. This is required because firestore doesn't accept undefined, only null/non null/missing value. If I were to define it as `string | null | undefined`, then I'd be forced to specify it, and possibly have it `undefined`.

TLDR; undefined isn't equal to missing a value.

File: functions/src/events-view/events-view-data.ts [hunk 2]
```diff
@@ -18,11 +18,11 @@ export interface Event {
...
+    track?: Track | null
```
Comment (fourlastor) [inline functions/src/events-view/events-view-data.ts]: small detail here: the question mark indicates that the value can not be there at all, instead of being undefined. This is required because firestore doesn't accept undefined, only null/non null/missing value. If I were to define it as `string | null | undefined`, then I'd be forced to specify it, and possibly have it `undefined`.

TLDR; undefined isn't equal to missing a value.
